modules/core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/impl/ExportedDocumentImpl.java

```diff
@@ -310,7 +310,7 @@ protected void readProperty(Element parent, Namespace targetNs, Field field, Obj
 
     protected final void readBlob(Element element, ComplexType ctype, Blob blob, boolean inlineBlobs)
             throws IOException {
-        String blobPath = Integer.toHexString(RANDOM.nextInt()) + ".blob";
+        String blobPath = blob.getFilename() != null ? blob.getFilename() : Integer.toHexString(RANDOM.nextInt()) + ".blob";
```

Comment:
With java 9 APIs this could be rewritten to:
```java
Objects.requireNonNullElseGet(blob.getFilename(), () -> Integer.toHexString(RANDOM.nextInt()));
```

Reply:
Or, simpler to read, using `StringUtils.defaultString` (when evaluating the second part is not too costly)